core/src/test/java/org/sql2o/converters/LocalDateConverterTest.java

```diff
@@ -0,0 +1,189 @@
+
+            query.executeUpdate();
+        }
+    }
+
+    private void insertInitialData() {
+        String sqlScriptOfInsertData = "insert into local_date_converter(created_at) values (:created_at)";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfInsertData)) {
+
+            query.addParameter("created_at", LocalDate.now());
+            query.executeUpdate();
+        }
+    }
+
+
+    @After
+    public void tearDown() {
+        this.dropTable();
+    }
+
+    private void dropTable() {
+        String sqlScriptOfDropTable = "drop table local_date_converter";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfDropTable)) {
+
+            query.executeUpdate();
+        }
+    }
+
+
+    @Test
+    public void whenFindAllLocalDates_thenTableHas1InitialRecordsOfLocalDateInstance() {
+
+        // When
+        List<LocalDate> localDates;
+        String sqlScriptOfFindAll = "select created_at from local_date_converter";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfFindAll)) {
+
+            List<MockLocalDate> mockLocalDates = query.executeAndFetch(MockLocalDate.class);
+            localDates = mockLocalDates.stream()
+                .map(MockLocalDate::getLocalDate)
+                .collect(Collectors.toList());
+        }
+
+        // Then
+        Assert.assertNotNull(localDates);
+        Assert.assertEquals(1, localDates.size());
+        Assert.assertEquals(LocalDate.class, localDates.get(0).getClass());
+    }
+
+
+    @Test
+    public void givenValidLocalDate_whenLocalDateSavedSuccessfully_thenTableHas2RecordsOfLocalDateInstance() {
+
+        // Given
+        LocalDate localDate = LocalDate.now();
+        String sqlScriptOfInsertData = "insert into local_date_converter(created_at) values (:created_at)";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfInsertData)) {
+            query
+                .addParameter("created_at", localDate)
+                .executeUpdate();
+        }
+
+        // When
+        List<LocalDate> localDates;
+        String sqlScriptOfFindAll = "select created_at from local_date_converter";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfFindAll)) {
+
+            List<MockLocalDate> mockLocalDates = query.executeAndFetch(MockLocalDate.class);
+
+            localDates = mockLocalDates.stream()
+                .map(MockLocalDate::getLocalDate)
+                .collect(Collectors.toList());
+        }
+
+        // Then
+        Assert.assertNotNull(localDates);
+        Assert.assertEquals(2, localDates.size());
+        Assert.assertEquals(LocalDate.class, localDates.get(0).getClass());
+        Assert.assertEquals(LocalDate.class, localDates.get(1).getClass());
+    }
```

Comment:
Is there any additional benefit to the test that fetches two rows compared to the test that examines only one row? At first glance, it appears that both tests evaluate the same thing. If I’m not mistaken, I recommend considering removing the test involving two rows.

Reply:
Thank you for your feedback.

The main purpose of checking two records here is to also verify the record initially added to the table by default. Because we don't know whether the upcoming data is new or old. That's why we are checking both records here.

However, if you have any other concerns or suggestions regarding the test coverage, I'm open to discussing them.

Reply:
But, both tests adds the date to the database using the exact same query and datatype. Therefore the second test, that fetches two rows, doesn't test any new variant of the converter. Yes, it tests that the Sql2o library can fetch more than one row, but there are plenty of tests testing that.

Reply:
In this case, we're not testing whether multiple lines are coming or not. Before running the tests, we create a table in the database and add a record to it. Later during the test, we add another record, and we need to check both if they are really added. I believe this approach is more accurate. If you don't think so, we can remove it. I don't think it's a critical situation, but during the testing phase, there will always be two rows in the table.